crontabber/tests/test_crontabber.py

```diff
@@ -2149,6 +2150,105 @@ def test_job_list_separated_sloppily(self):
             ok_('Ran FooJob' in infos, infos)
             ok_('Ran BarJob' in infos, infos)
 
+    def test_run_two_jobs_almost_simultaneously_first_time(self):
```

Comment:
I believe our instance of Postgres is run in "read committed" isolation mode. That means that we don't do read locking on transactions, only write locking.  That means a connection may experience non-repeatable reads within a transaction (read a column from a row once and see it is 0,  read it again later within the transaction and find it has been changed to 1000). 

The test uses two threads each having its own database connection.  There is nothing in the threading or transaction models to prevent this order of operations between the threads: 
```
01 thread1 starts
02 thread2 starts
03 thread1 sleep 0.1
04 thread2 sleep 0.05
05 thread2 app.py:969 read job X transaction begins & ends
06 thread1 app.py:959 read job X transaction begins & ends
07 thread1 app.py:1135 read job X transaction begins & ends
08 thread2 app.py:1135 read job X transaction begins & ends
09 thread1 app.py:286 begin transaction
10 thread1 app.py:294 read job X
11 thread2 app.py:286 begin transaction
12 thread2 app.py:294 read job X
13 # both thread 1 & 2 have the same info about unstarted job X
14 thread2 app.py:357 updates job X (locking job X row)
15 thread1 app.py:357 updates job X (BLOCKED)
16 thread2 app.py:361 commits transaction assumes control of job X
17 thread1 app.py:357 unblocked - updates job X
18 thread1 app.py:361 commits transaction assumes control of job X
19 # both threads now think they own job X and are running it.
```

Is this an unlikely scenario?  All I can say is that my intuition says to me that this wide open opportunity for a concurrency problem.  The ultimate solution is to use the NOWAIT as it would have caused _line 15_ above to FAIL unstead of BLOCK.